tests/Aspire.NATS.Net.Tests/AspireNatsClientExtensionsTests.cs

```diff
@@ -97,8 +113,111 @@ public void ConnectionNameWinsOverConfigSection(bool useKeyed)
             host.Services.GetRequiredKeyedService<INatsConnection>("nats") :
             host.Services.GetRequiredService<INatsConnection>();
 
-        Assert.Equal(ConnectionString, connection.Opts.Url);
+        Assert.Equal(_connectionString, connection.Opts.Url);
         // the connection string from config should not be used since it was found in ConnectionStrings
         Assert.DoesNotContain("unused", connection.Opts.Url);
     }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public async Task AddNatsClient_HealthCheckShouldBeRegisteredWhenEnabled(bool useKeyed)
+    {
+        var key = DefaultConnectionName;
+        var builder = CreateBuilder(_connectionString);
+
+        if (useKeyed)
+        {
+            builder.AddKeyedNatsClient(key, settings =>
+            {
+                settings.HealthChecks = true;
+            });
+        }
+        else
+        {
+            builder.AddNatsClient(DefaultConnectionName, settings =>
+            {
+                settings.HealthChecks = true;
+            });
+        }
+
+        var host = builder.Build();
+
+        var healthCheckService = host.Services.GetRequiredService<HealthCheckService>();
+
+        var healthCheckReport = await healthCheckService.CheckHealthAsync();
+
+        var healthCheckName = useKeyed ? $"NATS_{key}" : "NATS";
+
+        Assert.Contains(healthCheckReport.Entries, x => x.Key == healthCheckName);
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void AddNatsClient_HealthCheckShouldNotBeRegisteredWhenDisabled(bool useKeyed)
+    {
+        var builder = CreateBuilder(_connectionString);
+
+        if (useKeyed)
+        {
+            builder.AddKeyedNatsClient(DefaultConnectionName, settings =>
+            {
+                settings.HealthChecks = false;
+            });
+        }
+        else
+        {
+            builder.AddNatsClient(DefaultConnectionName, settings =>
+            {
+                settings.HealthChecks = false;
+            });
+        }
+
+        var host = builder.Build();
+
+        var healthCheckService = host.Services.GetService<HealthCheckService>();
+
+        Assert.Null(healthCheckService);
+    }
+
+    [RequiresDockerFact]
+    public void NatsInstrumentationEndToEnd()
+    {
+        RemoteExecutor.Invoke(async (connectionString) =>
+        {
+            var builder = CreateBuilder(connectionString);
+
+            builder.AddNatsClient(DefaultConnectionName, settings =>
+            {
+                settings.Tracing = true;
+            });
```

Comment:
```suggestion
            builder.AddNatsClient(DefaultConnectionName);
```

Tracing should be enabled by default.